### ISSUE TITLE
Error in SetKeyword when ft=racket

### ISSUE THREAD

**Issue by NieDzejkob** [kovisoft/paredit#37]
1. `vi test.rkt`
2. `i(<ESC>`
3. `,W`

Expected result: `(())`
Actual result:

```
Error detected while processing function PareditWrap[1]..<SNR>23_SetKeyword:
line    3:
E474: Invalid argument: iskeyword+=+,-,*,/,%,<,=,>,:,$,?,!,@-@,94,~,#,|,&
Press ENTER or type command to continue
```

This seems to be caused by repeated values, the racket plugin seems to set `iskeyword=@,48-57,_,192-255,#,%,^`.

**Comment by kovisoft** [comment]
Could you please tell me what racket plugin are you using?

**Comment by NieDzejkob** [comment]
Sorry for that! I've set it up so long ago that I assumed it was a part of the default vim installation...

It's https://github.com/wlangstroth/vim-racket

**Comment by kovisoft** [comment]
Strange, it seems that vim cannot append anything to `iskeyword` if it contains a `^` character.
Fixed in commit https://github.com/kovisoft/paredit/commit/462713f18748e392cf6860538a01c538a00a591f

**Comment by NieDzejkob** [comment]
Hmm, do you think this is intended behavior? Maybe it would be worth it to report a bug upstream?

Also, thanks! Your plugin is probably the one that impacts my Vim experience the most.

**Comment by kovisoft** [comment]
I'm not sure if this is a bug or something I'm not aware of. Anyway it's easy to reproduce:

```
:setlocal iskeyword=^
:setlocal iskeyword+=+
E474: Invalid argument: iskeyword+=+
```

**Comment by cwfoo** [comment]
@kovisoft The behavior for `^` (caret) is documented in `:help isfname`:

https://github.com/vim/vim/blob/v8.2.2556/runtime/doc/options.txt#L4514-L4519

```
	If a part starts with '^', the following character number or range
	will be excluded from the option.  The option is interpreted from left
	to right.  Put the excluded character after the range where it is
	included.  To include '^' itself use it as the last character of the
	option or the end of a range.  Example:
		"^a-z,#,^"	(exclude 'a' to 'z', include '#' and '^')
```

https://github.com/vim/vim/blob/v8.2.2556/runtime/doc/options.txt#L4529-L4531

```
	A comma can be excluded by prepending a '^'.  Example:
		" -~,^,,9"	All characters from space to '~', excluding
				comma, plus <Tab>.
```

**Comment by kovisoft** [comment]
Ah, thank you very much, I was not aware of this. Now I see what the real problem is. And I think my above fix is not fully okay, because I need to check for `^` that is the very last character of the option. Other `^` characters somewhere inside the option don't cause problems.

**Comment by cwfoo** [comment]
> Strange, it seems that vim cannot append anything to `iskeyword` if it contains a `^` character.
> Fixed in commit [462713f](https://github.com/kovisoft/paredit/commit/462713f18748e392cf6860538a01c538a00a591f)

https://github.com/kovisoft/paredit/blob/462713f18748e392cf6860538a01c538a00a591f/plugin/paredit.vim#L283-L290

Why is `setlocal iskeyword+=^` needed when `94` has already been added `iskeyword`?

**Comment by kovisoft** [comment]
You are absolutely right, thanks for all your comments. I removed the `setlocal iskeyword+=^`, see commit https://github.com/kovisoft/paredit/commit/d1176a7090bf9c1d9a9ca5bd12c6e859d5fe11d6
